ibis/expr/operations/relations.py

```diff
@@ -84,6 +84,48 @@ def change_name(self, new_name):
         return type(self)(new_name, self.args[1], self.source)
 
 
+@public
+class FileFormat(Node):
```

Comment:
I'm not sure we want these to be `Node`s vs `Annotable`s. @kszucs Thoughts?

Reply:
I'm really in favor of them being Node instances. This would generalize the table definitions for the file based backends.

```python

dataset = ibis.dataset("s3://bucket/<paritioning-info>/<glob-pattern>.parquet")  # or we could also overload ibis.table()
```

The underlying operation would be a `FileTable` with `PaquetFormat` and perhaps additional partitioning directives like `HivePartitioning` (this is similar how pyarrow datasets are defined). Then we could feed these tables to pyspark, datafusion, pyarrow, duckdb, clickhouse and possibly to other databases as well. 

Currently we need to define the source tables and the schemas for each individual backend using custom implementations in `ibis.<backend>.connect()`. IMO those should be moved to be expression level constructs.

Reply:
So instead of:

```python
client = ibis.pandas.connect(dictionary={'table': pd.read_csv(...)})
t = client['table']
```

it would be:

```python
t = ibis.dataset(pd.read_csv(...), schema=...)
# t.op() => ops.PandasTable extending ops.InMemoryTable
t = ibis.dataset(data_directory / 'table.csv', schema=...)
# t.op() => ops.FileTable 

expr = t['field'].sum()

ibis.pandas.execute(expr)
```

Reply:
I misunderstood your question, the format classes could indeed be Annotable subclasses rather than Node subclasses since these are not Operations.

Reply:
Thanks, I'll update this